lib/Byng/Pimcore/Sitemap/Plugin/Installer.php

```diff
@@ -0,0 +1,201 @@
+
+    /**
+     * Creates the redirect rules used to display the appropriate sitemap file
+     */
+    public function createRedirectRule()
+    {
+        $route = Staticroute::create();
+        $route->setName('sitemap')
+            ->setPattern('/^\/sitemap\.xml$/')
+            ->setReverse('/sitemap.xml')
+            ->setModule('PimcoreSitemapPlugin')
+            ->setController('sitemap')
+            ->setAction('view')
+            ->save();
+    }
+
+    /**
+     * Removes the redirect rule
+     */
+    public function deleteRedirectRule()
+    {
+        $route = Staticroute::getByName('sitemap');
+        $route->delete();
+    }
+
+    /**
+     * Creates the sitemap folder
+     */
+    public function createSitemapFolder()
+    {
+        // Create sitemap folder (if doesn't exist already)
+        if (!is_dir(SITEMAP_PLUGIN_FOLDER)) {
+            mkdir(SITEMAP_PLUGIN_FOLDER, 0777, true);
+        }
+    }
+
+    /**
+     * Deletes the sitemap folder and its children
+     */
+    public function deleteSitemapFolder()
+    {
+        if (is_dir(SITEMAP_PLUGIN_FOLDER)) {
+            if (!is_dir_empty(SITEMAP_PLUGIN_FOLDER)) {
+                array_map('unlink', glob(SITEMAP_PLUGIN_FOLDER . '/*'));
+            }
+            rmdir(SITEMAP_PLUGIN_FOLDER);
+        }
+    }
+
+    /**
+     * Creates the configuration file.
+     * @throws \Exception
+     */
+    public function createConfigFile()
+    {
+        // Get Sites FQDN map
+        $sites = $this->getSitesProtocolMap();
+
+        // Create xml config file
+        $config = new \Zend_Config_Xml(PIMCORE_PLUGINS_PATH . '/PimcoreSitemapPlugin/install/config.xml', null, ['allowModifications' => true]);
+        $config->sites = ['site' => $sites];
+
+        if (!is_dir(dirname(SITEMAP_CONFIGURATION_FILE))) {
+            if (!@mkdir(dirname(SITEMAP_CONFIGURATION_FILE), 0777, true)) {
+                throw new \Exception('Sitemap: Unable to create plugin config directory');
+            }
+        }
+
+        $configWriter = new \Zend_Config_Writer_Xml();
+        $configWriter->setConfig($config);
+        $configWriter->write(SITEMAP_CONFIGURATION_FILE);
+    }
+
+    /**
+     * Deletes the configuration file
+     */
+    public function deleteConfigFile()
+    {
+        if (file_exists(SITEMAP_CONFIGURATION_FILE)) {
+            unlink(SITEMAP_CONFIGURATION_FILE);
+        }
+    }
+
+    /**
+     * Lists the Pimcore sites and their properties (used to generate Config file)
+     * @return array
+     */
+    private function getSitesProtocolMap()
+    {
+        $client = new \Zend_Http_Client();
+        $sitesMap = [];
+
+        // Add the main domain
+        $defaultDomain = Config::getSystemConfig()->get("general")->get("domain");
+        $sitesMap[] = [
+            'rootId' => 1,
+            'rootPath' => '',
+            'protocol' => $this->getProtocolForDomain($defaultDomain, $client),
+            'domain' => $defaultDomain
+        ];
```

Comment:
Can the above array be wrapped in an if that checks if `$defaultDomain` is truthy? Otherwise it also errors here when installing if you've not set a general domain. I'd either do that, or make it so that it shows you a more specific error message telling you that you need to add that domain.